### PR TITLE
[Notifier] [FakeChat] Added the bridge

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | https://github.com/symfony/symfony-docs/pull/15171
| Recipe PR     | https://github.com/symfony/recipes/pull/928

This PR is based on https://github.com/symfony/symfony/pull/39949 but for chat messages instead of SMS.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -188,8 +189,6 @@
 use Symfony\Contracts\Translation\LocaleAwareInterface;
 
 /**
- * FrameworkExtension.
```
Comment (Nyholm) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: This comment has been around for over a decade..

Im not against it... Im just saying.

File: src/Symfony/Component/Notifier/Bridge/FakeChat/FakeChatTransportFactory.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        }
+
+        if ('fakechat+email' === $scheme) {
+            $serviceId = $dsn->getHost();
+            $to = $dsn->getRequiredOption('to');
+            $from = $dsn->getRequiredOption('from');
+
+            return (new FakeChatEmailTransport($this->serviceProvider->get($serviceId), $to, $from))->setHost($serviceId);
+        }
+    }
```
Comment (Nyholm) [inline src/Symfony/Component/Notifier/Bridge/FakeChat/FakeChatTransportFactory.php:52]: Could we simplify the logic of this method?

**Review by Nyholm — COMMENTED** [review]
Thank you

File: src/Symfony/Component/Notifier/Bridge/FakeChat/FakeChatTransportFactory.php (hunk 2 above)
Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/FakeChat/FakeChatTransportFactory.php:52]: For now I just copied it from `FakeSmsTransportFactory`. Any proposal?

Comment (Nyholm) [reply src/Symfony/Component/Notifier/Bridge/FakeChat/FakeChatTransportFactory.php:52]: Yes!

See https://github.com/symfony/symfony/pull/39949#issuecomment-809320955

Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/FakeChat/FakeChatTransportFactory.php:52]: Ah yes, but we plan to add 2 more transports, so we would again end up with sth like this

**Comment by nicolas-grekas** [comment]
`LICENSE` and `.git*` files are missing

**Comment by OskarStark** [comment]
> LICENSE and .git* files are missing

Thanks, fixed

**Comment by OskarStark** [comment]
Rebased

File: src/Symfony/Component/Notifier/Bridge/FakeChat/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+Fake Chat Notifier
+==================
+
+Provides Fake CHAT (as email during development) integration for Symfony Notifier.
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/FakeChat/README.md]:
```suggestion
Provides Fake Chat (as email during development) integration for Symfony Notifier.
```

**Comment by OskarStark** [comment]
I merge this, as its mostly a copy paste from `FakeSms` bridge

File: src/Symfony/Component/Notifier/Bridge/FakeChat/FakeChatTransportFactory.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+use Symfony\Contracts\Service\ServiceProviderInterface;
+
+/**
+ * @author Oskar Stark <oskarstark@googlemail.com>
+ */
+final class FakeChatTransportFactory extends AbstractTransportFactory
+{
+    protected $serviceProvider;
+
+    public function __construct(ServiceProviderInterface $serviceProvider)
```
Comment (chalasr) [inline src/Symfony/Component/Notifier/Bridge/FakeChat/FakeChatTransportFactory.php:27]: Same as #40731, where is this locator configured? The framework integration seems broken

**Pull request merged**